src/platform/Makefile.am

```diff
@@ -187,13 +190,50 @@ endif # CHIP_DEVICE_LAYER_TARGET_DARWIN
 
 if CHIP_DEVICE_LAYER_TARGET_LINUX
 
+libDeviceLayer_a_CPPFLAGS              += \
+    -I$(builddir)/gen                     \
+    $(NULL)
+
 SUBDIRS                                += tests
 
+.PHONY: gen
+gen:
+	mkdir -p gen
+
+BUILT_SOURCES                         = \
+    gen                                 \
+    $(NULL)
+
 libDeviceLayer_a_CPPFLAGS              += \
     $(GIO_CFLAGS)                         \
+    $(GIO_UNIX_CFLAGS)                    \
     $(INIPP_CPPFLAGS)                     \
     $(NULL)
 
+if CONFIG_BLE_PLATFORM_BLUEZ
+
+BUILT_SOURCES                         += \
+    gen/DbusBluez.h                      \
+    gen/DbusBluez.c                      \
+    $(NULL)
+
+gen/DbusBluez.h: Linux/org.bluez.xml
+	gdbus-codegen --header --output $@ --c-namespace CHIPDBus $<
+
+gen/DbusBluez.c: Linux/org.bluez.xml
+	gdbus-codegen --body --output $@ --c-namespace CHIPDBus $<
+	sed -i 's/config\.h/BuildConfig.h/g' $@
+
+nodist_libDeviceLayer_a_SOURCES        += \
```

Comment:
gen/DbusBluez.c has already been added to libDeviceLayer_a_SOURCES, why adding it to nodist_libDeviceLayer_a_SOURCES?

Reply:
yep, just removed.